lib/src/util/posix/daemonize.cc

```diff
@@ -242,9 +242,15 @@ std::unique_ptr<PIDFile> daemonize() {
 
     // Redirect standard files; we always use boost::log anyway
 
-    stdin = freopen("/dev/null", "r", stdin);
-    stdout = freopen("/dev/null", "w", stdout);
-    stderr = freopen("/dev/null", "w", stderr);
+    if (freopen("/dev/null", "r", stdin) == nullptr) {
+        LOG_WARNING("Failed to redirect stdin to /dev/null; errno=%1%", std::to_string(errno));
```

Comment:
Ohhh. I didn't look around :)

We can save that for a later PR that normalizes the errno reporting.